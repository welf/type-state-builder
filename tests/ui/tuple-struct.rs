use type_state_builder::TypeStateBuilder;

// This should be an error: tuple structs are not supported
#[derive(TypeStateBuilder)]
struct TupleStruct(String, u32);

fn main() {}
