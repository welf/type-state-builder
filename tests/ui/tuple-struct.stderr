error: TypeStateBuilder only supports structs with named fields
 --> tests/ui/tuple-struct.rs:5:1
  |
5 | struct TupleStruct(String, u32);
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no function or associated item named `builder` found for struct `TupleStruct` in the current scope
 --> tests/ui/tuple-struct.rs:8:34
  |
5 | struct TupleStruct(String, u32);
  | ------------------ function or associated item `builder` not found for this struct
...
8 |     let _instance = TupleStruct::builder()
  |                                  ^^^^^^^ function or associated item not found in `TupleStruct`
